code/game/gamemodes/survivor.dm

```diff
@@ -0,0 +1,460 @@
+        var/ratio = rand(0, 100)
+        switch (player_size)
+            if(HIGH_POP)
+                ratio = rand(5,50)
+            if(MED_POP)
+                ratio = rand(25,50)
+            if(LOW_POP)
+                ratio = rand(50,100)
+        A.cell.charge = A.cell.maxcharge * (ratio / 100)
+    */
+
+    for (var/I in GLOB.machines)
+        if (istype(I, /obj/machinery/door))
+            var/obj/machinery/door/airlock/A = I
+            A.req_access = null
+            A.req_access_txt = null
+            A.req_one_access = null
+            A.req_one_access_txt = null
+
+        // Charge every apc across the map
+        if (isAPC(I))
+            var/obj/machinery/power/apc/A = I
+            var/ratio = rand(5,50)
+            A.cell.charge = A.cell.maxcharge * (ratio / 100)
+
+    return TRUE
+
+
+/datum/game_mode/survivor/proc/transform_xeno(datum/mind/M)
+    var/mob/living/carbon/Xenomorph/Larva/X = new (pick(GLOB.xeno_spawn))
+
+    if(isnewplayer(M.current))
+        var/mob/new_player/N = M.current
+        N.close_spawn_windows()
+        N.new_character = X
+
+    M.transfer_to(X, TRUE)
+
+    to_chat(X, "<B>You are now an alien!</B>")
+    to_chat(X, "<B>Your job is to spread the hive and protect the Queen. If there's no Queen, you can become the Queen yourself by evolving into a drone.</B>")
+    to_chat(X, "Talk in Hivemind using <strong>;</strong> (e.g. ';My life for the queen!')")
+
+    X.update_icons()
+
+
+/datum/game_mode/survivor/proc/transform_queen(datum/mind/M)
+    // var/mob/living/carbon/Xenomorph/Queen/X = new (pick(GLOB.survivor_spawn_xeno))
+    var/mob/living/carbon/Xenomorph/Queen/X = new (pick(GLOB.xeno_spawn))
+
+    if(isnewplayer(M.current))
+        var/mob/new_player/N = M.current
+        N.close_spawn_windows()
+        N.new_character = X
+
+    M.transfer_to(X, TRUE)
+
+    to_chat(X, "<B>You are now the alien queen!</B>")
+    to_chat(X, "<B>Your job is to spread the hive.</B>")
+    to_chat(X, "Talk in Hivemind using <strong>;</strong> (e.g. ';My life for the hive!')")
+
+    X.update_icons()
+
+
+/datum/game_mode/survivor/proc/spawn_queen_in_ovi()
+    var/mob/living/carbon/Xenomorph/Queen/Q = new (pick(GLOB.xeno_spawn))
+    Q.ovipositor = TRUE
+    Q.away_time = 0 // queen so they can't be taken by late joins
+
+
+/datum/game_mode/survivor/proc/transform_survivor(datum/mind/M, late_join = FALSE,)
+    var/spawn_point = pick(GLOB.survivor_spawn_human)
+    if (late_join)
+        var/mob/A = pick(GLOB.alive_human_list)
+        if(A && !isspaceturf(get_turf(A.loc)))
+            spawn_point = A.loc
+
+    var/mob/living/carbon/human/H = new(spawn_point)
+
+    if(isnewplayer(M.current))
+        var/mob/new_player/N = M.current
+        N.close_spawn_windows()
+        N.new_character = H
+
+    M.transfer_to(H, TRUE)
+    H.client.prefs.copy_to(H)
+
+    var/job_title
+    var/role_guide
+    var/survivor_job = pick(subtypesof(/datum/job/survivor))
+
+    switch(rand(0, 100))
```

Comment:
Why not 1 to 4?

Reply:
Because then you wouldn't have something to review (i fucked up)